shell/browser/electron_permission_manager.h

```diff
@@ -81,6 +86,17 @@ class ElectronPermissionManager : public content::PermissionControllerDelegate {
                                   const GURL& requesting_origin,
                                   const base::DictionaryValue* details) const;
 
+  bool CheckDevicePermission(content::PermissionType permission,
+                             content::WebContents* web_contents,
+                             const url::Origin& origin,
+                             const base::Value* object,
+                             content::RenderFrameHost* render_frame_host) const;
```

Comment:
nit: let's only pass `RenderFrameHost` and use `WebContents::FromRenderFrameHost` to fetch the `WebContents`. There's no reason to pass a `WebContents*` other than the one that `WebContents::FromRenderFrameHost` would return, so this eliminates a potential source of error (passing a mismatching WebContents/RFH pair).

```suggestion
  bool CheckDevicePermission(content::PermissionType permission,
                             content::RenderFrameHost* render_frame_host,
                             const url::Origin& origin,
                             const base::Value* object) const;
```